### ISSUE TITLE
Make it work offline

### ISSUE THREAD

**Issue by stefanjudis** [stefanjudis/perf-tooling#778]
I'd like to play around with Service Workers.

What do you think, wanna implement it here?

**Comment by marcobiedermann** [comment]
I think we would need a SSL certificate for that. But I am 100% pro this implementation.

**Comment by stefanjudis** [comment]
@marcobiedermann can you point me somewhere proving that?

Never heard that TLS is mandatory for Service Workers.

**Comment by ben-eb** [comment]
http://www.w3.org/TR/service-workers/#security-considerations

**Comment by stefanjudis** [comment]
Hah! 

Well then let's do it. :)

thanks @ben-eb for shout out.

**Comment by rupl** [comment]
This thing is hosted on Heroku, right? SSL is $20/month, not a terrible cost. I'd donate a few months' worth to the cause :bowtie: Or does Fastly have some option they could enable, since they are your generous sponsor at the moment?

Either way, the site is screaming for SW treatment, and it appears that the current architecture (most interactions are client-side) means SW would be a snap. One load and the app is installed, then the client-side search takes over from there.

Actually following links is another story. But still, it would be possible to browse the lists offline, maybe mark them for consumption when connection returns?

**Comment by stefanjudis** [comment]
hey hey. 

It's already on a payed heroku account. ;)

Yeah man, I'd love to have that ( and to start playing with SW ).
Time is the issue here. ;)

**Comment by rupl** [comment]
yeah man! I am unemployed right now so I have time to play around. I've never actually deployed any SW stuff but you gotta start somewhere

**Comment by stefanjudis** [comment]
Well I'd love to give you access and/or play around with this. ;)

**Comment by rupl** [comment]
excellent! and the HTTPS issue wont block development since `localhost` is a valid SW host

**Comment by stefanjudis** [comment]
:+1:

**Comment by marcobiedermann** [comment]
@stefanjudis [CloudFlare](https://www.cloudflare.com/) offers a free SSL certificate. It's a one click setup.  Just create a free account and add your domain. Change SSL connection to "flexible" and set the page rule to "always use HTTPS". This should take around 24h to kick in.

**Comment by stefanjudis** [comment]
Then still I have to configure heroku to use this and pay for it. ;)

Also I'd like to use https://letsencrypt.org/ instead. :)

**Comment by rupl** [comment]
I've been tinkering on my own site for the last two days and think I will have a first pass for this site soon. It can be tested SSL-free on `localhost` hostname.

**Comment by stefanjudis** [comment]
:+1: @rupl 

Let me know when I have to hurry up with the TLS. :)

**Comment by rupl** [comment]
small update: the initial SW is a snap to put together, but I'm still trying stuff out on my site. Busting the cache is important for a content-driven site like this one so I want to get it working smooth before we roll it out at all.

**Comment by stefanjudis** [comment]
Sounds perfect!
